Update footer styles in multi page quiz

Imran92 opened the pull request:
Resolves https://github.com/Automattic/sensei/issues/7243

## Proposed Changes
* Updated styles for multi-paged quizzes in learning mode

## Testing Instructions
<!--
Add as many details as possible to help others reproduce the issue and test the changes.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

1. Add a multi-page quiz to a lesson.
2. Select the Allow Retakes checkbox.
3. Checkout this branch of Course theme https://github.com/Automattic/themes/pull/7462
4. As a student, view the quiz on the frontend.
5. Scroll to the footer.
6. Make sure the design matches the new structure of footers across LM
7. Make sure normal single page footer is not broken
8. Make sure lesson footer is not broken
9. Check in mobile view
10. Check for all variations of course theme and other top used themes

#### Course theme
![Screenshot 2023-11-06 at 3 55 49 PM](https://github.com/Automattic/sensei/assets/6820724/b0992cb5-f251-4970-8ead-37db9223a6ee)
![Screenshot 2023-11-06 at 3 56 20 PM](https://github.com/Automattic/sensei/assets/6820724/058e58c3-0e07-4093-8ab5-97d5ed9c315d)

#### Astra
![Screenshot 2023-11-06 at 3 56 48 PM](https://github.com/Automattic/sensei/assets/6820724/1e4574a4-7eac-445f-8eb1-502dba8fcc73)

## Pre-Merge Checklist
<!-- Complete applicable items on this checklist **before** merging. Items that are not applicable can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed. -->
- [x] PR title and description contain sufficient detail and accurately describe the changes
- [x] Acceptance criteria is met
- [ ] Decisions are publicly documented
- [x] Adheres to coding standards ([PHP](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/), [JavaScript](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/), [CSS](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/css/), [HTML](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/html/))
- [ ] All strings are translatable (without concatenation, handles plurals)
- [ ] Follows our naming conventions (P6rkRX-4oA-p2)
- [ ] Hooks (p6rkRX-1uS-p2) and functions are documented
- [x] New UIs are responsive and use a [mobile-first approach](https://zellwk.com/blog/how-to-write-mobile-first-css/)
- [x] New UIs match the designs
- [ ] Different user privileges (admin, teacher, subscriber) are tested as appropriate
- [ ] Code is tested on the minimum supported PHP and WordPress versions
- [ ] User interface changes have been tested on the latest versions of Chrome, Firefox and Safari
- [ ] "Needs Documentation" label is added if this change requires updates to documentation
- [ ] Known issues are created as new GitHub issues

Imran92 committed:
Update button texts

Imran92 committed:
Fix button width

Imran92 committed:
Change button positions

Imran92 committed:
Fix colors for the link buttons

Imran92 committed:
Fix sizes for course theme variations

Imran92 committed:
Fix lateral padding on mobile view

Imran92 committed:
Add changelog

Imran92 committed:
Update complete quiz text

Imran92 committed:
Fix outline button not taking full space in mobile view

Imran92 committed:
Fix button width in mobile view

Imran92 committed:
Fix icon alignment in quiz notice

Imran92 committed:
Fix next lesson button color

Imran92 committed:
Fix footer for multiple button, match sizes

Imran92 committed:
Remove unnecessary class

Imran92 committed:
Remove unnecessary spacing

Imran92 committed:
Fix previous and complete button sizings in paginated quiz

Imran92 committed:
Change alignment in mobile view

Imran92 committed:
Fix button sizings

Imran92 committed:
Fix button sizings in footer

Imran92 commented:
Having only "Next" as the button text does not look good to me for some reason, should we change this to "Next Questions" or "Next Page" or something like that? In the context of Lessons, we use "Next Lesson" as button text

donnapep commented:
> Having only "Next" as the button text does not look good to me for some reason, should we change this to "Next Questions" or "Next Page" or something like that? In the context of Lessons, we use "Next Lesson" as button text

If we use "Next Lesson" for lessons, then "Next Question" makes the most sense here. 👍🏻

Imran92 committed:
Update next and previous button text

Imran92 commented:
> If we use "Next Lesson" for lessons, then "Next Question" makes the most sense here. 👍🏻

Thank you, I've updated the Next and Previous buttons here https://github.com/Automattic/sensei/pull/7268/commits/8a9a0e9dbc9509dfb80a250b989da8f1bc52fd42, now they say "Previous Question" and "Next Question"

merkushin reviewed (commented):
Looks good to me, in general. Tested it when the quiz was not submitted and when it was submitted: both looks good.

I found an edge case, which is a bit broken. I don't know if we want to address it here (or at all). For this reason I don't approve the PR, let me know if we're fine to merge it as is and I approve it.

Now, to the edge case. It happens when the student completed the lesson, then the teacher adds a quiz to that lesson. When I open the quiz as a student I see it like that (pagination links are smaller, the "Next Question" button is in a different place):
![CleanShot 2023-11-07 at 16 59 11@2x](https://github.com/Automattic/sensei/assets/329356/9a17b839-9363-46d1-8055-95290978a06b)

At the same time, we don't need pagination here: all questions are here.

merkushin commented:
And as for the "Next Question"/"Next Page" discussion.
It is possible to have more than one question per page, so "Next Question" (singular) looks not correct here, I think.
In the screenshot, you can see I am on the second page and I have two questions per page, but I see "Previous Question" on the button. I would expect "Previous Questions" in this case.
![CleanShot 2023-11-07 at 17 12 10@2x](https://github.com/Automattic/sensei/assets/329356/12797c6d-07dd-467e-9f47-a064850f24a9)

Imran92 committed:
Show questions if we have plural count

Imran92 commented:
Hi @merkushin 👋 Thanks for taking a look.

> I found an edge case, which is a bit broken. I don't know if we want to address it here (or at all). For this reason I don't approve the PR, let me know if we're fine to merge it as is and I approve it.
> 
> Now, to the edge case. It happens when the student completed the lesson, then the teacher adds a quiz to that lesson. When I open the quiz as a student I see it like that (pagination links are smaller, the "Next Question" button is in a different place):

I've investigated the issue to find the root cause, found the cause after deep investigation, I'm sharing it in details because I felt a bit concerned about it as it's happening due to some changes related to HPPS and you may need to look for and handle potential similar issues in other places as well if they exist.

In learning mode, when viewing a quiz, we use the lesson or the quiz template conditionally. If the quiz is not taken by the student, we load the quiz template, and if the quiz is in awaiting-grade or any later status, we use the lesson template. you can find the logic [here](https://github.com/Automattic/sensei/blob/3e40b408aecb464c9e61ebda874a95c8cda1fdcc/includes/course-theme/class-sensei-course-theme-templates.php#L157-L173).

The problem is, for HPPS, I see that a change was introduced here and many other places recently to read quiz/lesson status from https://github.com/Automattic/sensei/pull/7185/files quiz repository instead. Previously, when a quiz was not taken by a student but the lesson was marked as completed, the status we got in comment was 'completed', but quiz repository is returning "passed" in that case, which is not correct because the student did not actually pass the quiz. I reverted the repository change temporarily to check if it works as expected, and it did. So I think we should handle the status issue in the HPPS project. I think it probably is an important issue because it is impacting not only the multi page quizzes, but other normal quizzes too and apart from the LM template loading, it can have impact in other places as well.

> I would expect "Previous Questions" in this case.

Updated here for Next and Previous Questions https://github.com/Automattic/sensei/pull/7268/commits/f71c93a5bc1ae23c635e77c21c4920c0e39954ff 👍

merkushin commented:
Thanks for investigation @Imran92!
You're right, the template was determined wrongly because of those changes.

Just to inform I want to notice, that we can't return 'completed' status for a quiz progress as it breaks the logic ('completed' is not a quiz progress status, that's a lesson progress status) and will break compatibility between comments-based and tables-based versions.
However, if you notice a similar issue, please ping me or Miro for sooner resolution. I fixed this issue and a similar one in this [PR](https://github.com/Automattic/sensei/pull/7278) (it's not open yet as I want to add some tests).

Also I addressed this issue there:
> At the same time, we don't need pagination here: all questions are here.

It wasn't related to HPPS project, but highlights the problem of unified progress for a lesson and a quiz.
In fact we need pagination there, but also have to output questions paginated.

Imran92 commented:
Hi @merkushin 👋 Thanks for the review!

> Just to inform I want to notice, that we can't return 'completed' status for a quiz progress as it breaks the logic ('completed' is not a quiz progress status, that's a lesson progress status) and will break compatibility between comments-based and tables-based versions.

Yap, fully agreed.  The problem was it was returning "Passed" which is a quiz-specific one. We just needed it to be something other than these 4 quiz-specific ones `[ 'ungraded', 'graded', 'passed', 'failed' ]`. Thanks for handling it now!

> However, if you notice a similar issue, please ping me or Miro for sooner resolution

Sure I will, thanks!

I'm not merging this PR yet as it depends on this PR https://github.com/Automattic/sensei/pull/7256.

Imran92 commented:
Hi @merkushin 👋 Sorry the review got dismissed after the base branch was merged to trunk, can you kindly do it again?

Thanks!!

Imran92 committed:
Merge branch 'trunk' into fix/button-styles-for-multi-page-quiz
